tests/Unit/Utilities/CssDocumentTest.php

```diff
@@ -0,0 +1,403 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Tests\Unit\Utilities;
+
+use Pelago\Emogrifier\Utilities\CssDocument;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \Pelago\Emogrifier\Utilities\CssDocument
+ */
+final class CssDocumentTest extends TestCase
+{
+    /**
+     * @var string
+     */
+    private const VALID_AT_FONT_FACE_RULE = '
+        @font-face {
+          font-family: "Foo Sans";
+          src: url("/foo-sans.woff2") format("woff2");
+        }
+    ';
+
+    /**
+     * @test
+     *
+     * @param string $selector
+     *
+     * @dataProvider provideSelector
+     */
+    public function parsesSelector(string $selector): void
+    {
+        $css = $selector . '{ color: green; }';
+        $subject = new CssDocument($css);
+
+        $result = $subject->getStyleRulesData([]);
+
+        self::assertCount(1, $result);
+        $trimmedResultantSelector = \trim($result[0]['selectors']);
+        self::assertSame(\trim($selector), $trimmedResultantSelector);
```

Comment:
We the `assertSame` combined with a `trim` multiple times. Maybe move this to a dedicated `assert` function? (Let's keep it here in this test. We can factor it out when we need it in multiple test cases.)

Reply:
Added a static method `assertSameTrimmed`.

There's also another assertion using `trim` but with an `assertThat`, which can't use this - it would need a `Constraint` child class to provide the functionality (along with a `TestCase` for any such new class).  I propose not implementing such right now, as the requirement may change if/when the Sabberworm library is introduced, as this is liable to result in additional whitespace differences which need to be accounted for.